Blog post: Astro dev with DDEV article (#116)
ddev/ddev.com: src/content/blog/astro-development-with-ddev.md

---
title: "Astro development with DDEV"

> pubDate: 2023-11-30

Reviewer: ```suggestion
pubDate: 2023-12-01
```

Maybe :) But we're going to get it!

# modifiedDate: 2023-07-23

> summary: How to run Astro, Vite and other npm packages with DDEV

Reviewer: ```suggestion
summary: How to run DDEV.com locally using DDEV, Astro and Vite
```

author: Bernardo Martinez
featureImage:
    src: /img/blog/2023/10/astroonddev.png
    alt: Astro image
categories:
  - Guides
  - DevOps
---

> 

Reviewer: Please add an introduction here that explains what the article is about.

> Some projects might require you to add front-end tooling. Or the whole project might be based around client-side frameworks. DDEV provides an npm command out of the box, simplifying Docker development. I will be using this website as an example.

Reviewer: I don't think this article is about DDEV having an `npm` command, lead with a stronger paragraph.

> DDEV.com is an [Astro](https://astro.build/) website. Astro is fantastic tool that packed with features that provide a low-entry barrier for developers looking to contribute features, issues and more. Among it's highlights are its low hosting cost, file-based routing, and extensive Markdown support.

Reviewer: ```suggestion
DDEV.com is an [Astro](https://astro.build/) website. Astro is fantastic static website tool that is packed with features that provides a low-entry barrier for developers looking to contribute features, issues and more. Among its highlights are its low hosting cost, file-based routing, and extensive Markdown support.
```

> That said, for a while this website did not use DDEV. Adding DDEV makes it easier for those with the tool to contribute PRs, but also provides an example on how other projects can take advantage of it.

Reviewer: It's unclear what we're talking about here. Strengthen this paragraph.

> As of this article, there are two ways to setup your local development on DDEV.com. With DDEV and without DDEV. Whichever is easier and more convenient for you.

Reviewer: ```suggestion
Now there are two ways to set up local development for DDEV.com. You can do it with `npm` on the host, or use DDEV itself.
```

I suspect the article should only be about using it with DDEV, and a link to the README will show other things.

> This guide will highlight what was required to add DDEV to an Astro project. It's meant to provide you with some background as well as tips along the way. That said, feel free to look at this [website](https://github.com/ddev/ddev.com/blob/main/.ddev/config.yaml) config first and reverse engineer your approach if that's easier.

Reviewer: If the article is about how to use DDEV with Astro, it would be better to just add a step-by-step at the end saying how. People don't need to look at the config.yaml though.

It's worth mentioning Astro uses Vite as its bundler so these steps would be very similar for any projects that might use Vite.

The instructions below assume you have DDEV and Docker installed and globally available.

> ## 1. Run `ddev config`

Reviewer: Take a look at the local preview to see that this isn't working out - you can't use `## 1.` in both these places, doesn't do anything.

For the purposes of ddev.com the project name could be left as the default as that will inherit the name of the folder ddev is in.

Astro by default saves it builds on the dist directory so our docroot/web folder was set to dist. This will allow you to browse the build on the default https://<projectname>.ddev.site URL.

DDEV currently uses the PHP project type as the custom catch-it-all. So for anything that does not fit inside the other CMSs PHP can be chosen.

## 1. Vite in DDEV

> There are a couple of ways one could run Vite in DDEV.

Reviewer: Just say what you did, don't talk about a couple of ways.
